Detectors/EMCAL/simulation/src/LZEROElectronics.cxx

```diff
@@ -0,0 +1,241 @@
+  } else if (TimeSums[3] < TimeSums[2] && TimeSums[2] < TimeSums[1] && TimeSums[1] >= TimeSums[0]) {
+    trendOfDigitsInTower = true;
+  } else if (TimeSums[0] > 0. || TimeSums[1] > 0. || TimeSums[2] > 0. || TimeSums[3] > 0.) {
+  }
+  double integralOfADCvalues = 0;
+  for (auto it = TimeSums.begin(); it != TimeSums.end(); it++) {
+    integralOfADCvalues += *it;
+  }
+  bool peakOverThreshold = false;
+  if (integralOfADCvalues > mThreshold && trendOfDigitsInTower) {
+    peakOverThreshold = true;
+  }
+  return peakOverThreshold;
+}
+//_____________________________________________________________________
+// Peak finding algorithm on all patches
+// It fills the mPeakFound vector with potential 1s
+bool LZEROElectronics::peakFinderOnAllPatches(Patches& p)
+{
+  bool isFoundGlobal = false;
+  p.mFiredPatches.clear();
+  for (auto& patches : p.mIndexMapPatch) {
+    auto PatchID = std::get<0>(patches);
+    auto isFound = peakFinderOnPatch(p, PatchID);
+    if (isFound) {
+      p.mFiredPatches.push_back(PatchID);
+      isFoundGlobal = true;
+    }
+  }
+  return isFoundGlobal;
+}
+//________________________________________________________
+void LZEROElectronics::init()
+{
+  mSimParam = &(o2::emcal::SimParam::Instance());
+  mRandomGenerator = new TRandom3(std::chrono::high_resolution_clock::now().time_since_epoch().count());
+  mSimulateNoiseDigits = mSimParam->doSimulateNoiseDigits();
+  mTriggerMap = new TriggerMappingV2(mGeometry);
+  setThreshold(mSimParam->getThresholdLZERO());
+  // setThreshold(132.);
+}
+//________________________________________________________
+void LZEROElectronics::clear()
+{
+}
+//________________________________________________________
+void LZEROElectronics::updatePatchesADC(Patches& p)
+{
+  p.updateADC();
+}
+//_______________________________________________________________________
+void LZEROElectronics::addNoiseDigits(Digit& d1)
+{
+  double amplitude = d1.getAmplitude();
+  double sigma = mSimParam->getPinNoiseTRU();
+  // double sigma = 0.04;
+
+  uint16_t noise = std::floor(std::abs(mRandomGenerator->Gaus(0, sigma) / constants::EMCAL_TRU_ADCENERGY)); // ADC
+
+  Digit d(d1.getTower(), 0., noise, d1.getTimeStamp());
+
+  d1 += d;
+}
+//_______________________________________________________________________
+void LZEROElectronics::fill(const std::deque<o2::emcal::DigitTimebinTRU>& digitlist, const o2::InteractionRecord record, std::vector<Patches>& patchesFromAllTRUs)
+{
+  int counterDigitTimeBin = 0;
+  int sizemDigitMap = -999;
+
+  for (auto& digitsTimeBin : digitlist) {
+    // Inside the DigitTimebinTRU
+    // Fill the LZEROElectronics with the new ADC value
+    // At the end of the loop run the peak finder
+    // Ship to LONEElectronics in case a peak is found
+    // Entire logic limited to timebin by timebin -> effectively implementing time scan
+
+    counterDigitTimeBin++;
+
+    int counterhelp = 0;
```

Comment:
Please use more meaningful names than `counterhelp` (what does `help` mean?)

Reply:
Removed, it was needed for an earlier check